### ISSUE TITLE
Precision and recall is very low

### ISSUE THREAD

**Issue by dashrathc** [Russell91/TensorBox#144]
@Russell91 

We have trained TensorBox on our custom dataset for 800k iterations with checkpointing every 50k iterations. We have evaluated trained model using test data. The precision is 0.21 and recall is 0.82, here precision is very low which is unexpected. The same run using ApolloCaffe's version gives 0.88 precision and 0.83 recall.

Following are the evaluation graph,
1) 800K run
![results-800k](https://cloud.githubusercontent.com/assets/7776101/21048502/b37c64d0-be35-11e6-8d3d-8b5a262bb657.png)

2) 400k run
![results-400k](https://cloud.githubusercontent.com/assets/7776101/21048522/bd82efee-be35-11e6-9d61-06c9a9e1123f.png)

As you can see that both the graphs look same though there is a difference in a number of iterations used for training.

@Russell91 can you please share your thoughts and help out to understand it?

**Comment by Russell91** [comment]
It appears that the highest precision is 89, and highest recall is 83 in these charts, no? Can you share which json file for the model you are using as well?

**Comment by dashrathc** [comment]
@Russell91 Thanks for your quick response.

Following is the JSON file content,
{
    "data": {
        "train_idl": "/Reinspect_Tensorbox/TensorBox/data/train.json",
        "test_idl": "/Reinspect_Tensorbox/TensorBox/data/test.json",
        "truncate_data": false
    },
    "logging": {
        "display_iter": 50,
        "save_iter": 50000
    },
    "solver": {
        "opt": "RMS",
        "use_jitter": false,
        "rnd_seed": 1,
        "epsilon": 0.00001,
        "learning_rate": 0.001,
        "learning_rate_step": 33000,
        "hungarian_iou": 0.5,
        "weights": "",
        "head_weights": [1.0, 0.1]
    },
    "use_lstm": true,
    "use_rezoom": true,
    "biggest_box_px": 10000,
    "rezoom_change_loss": "center",
    "rezoom_w_coords": [-0.25, 0.25],
    "rezoom_h_coords": [-0.25, 0.25],
    "reregress": true,
    "focus_size": 1.8,
    "early_feat_channels": 256,
    "later_feat_channels": 832,
    "avg_pool_size": 5,
    "num_lstm_layers": 2,
    "image_width": 640,
    "image_height": 480,
    "grid_height": 15,
    "grid_width": 20,
    "batch_size": 1,
    "region_size": 32,
    "clip_norm": 1.0,
    "lstm_size": 500,
    "deconv": false,
    "num_classes": 2,
    "rnn_len": 5
}

Also, one more question is why Precision suddenly decreases?
Would appreciate your help in this regard.

**Comment by piyushshri** [comment]
@Russell91 Can you please elaborate how significant is "highest precision" and "highest recall"? To measure the accuracy of the model, should which values of precision and recall should we consider amongst the "highest" and the "latest", and why?

**Comment by pmann88** [comment]
@Russell91 How to calculate average precision from the above PR curve or log average miss rate? These single numeric values will be easy to interpret and to use as performance measure when want to compare the performance with other algorithms. Thanks.
